### PR TITLE
ES6-ify codebase.

### PR DESCRIPTION
**Pull request by XhmikosR**
Also:

* remove unused variables
* move `require`s at the top
* check for prototype built-ins
* run in strict mode
* always check for error values in tests
* make style consistent
* mark Node.js 6 as the minimum supported version in package.json

---

This is a great improvement IMO, but it will require the use of proper linting, which is something the maintainers should choose :)

That being said, this can be reviewed more easily without the whitespace changes, see <https://github.com/nickmerwin/node-coveralls/pull/238/files?w=1>.

Everything should work fine on Node.js >= 6. Here's a test with Node.js 6.0.0: https://github.com/XhmikosR/node-coveralls/runs/276028000#step:7:105

### CODE REVIEW

File: bin/coveralls.js [hunk 1]
```diff
@@ -1,22 +1,23 @@
 #!/usr/bin/env node
-var handleInput = require('../lib/handleInput');
-var logger = require('../lib/logger');
```
Comment (XhmikosR) [inline bin/coveralls.js]: This seems to be unused. If not let me know and I'll change it to just `require('../lib/logger')`

File: lib/convertLcovToCoveralls.js [hunk 2]
```diff
@@ -1,90 +1,103 @@
-var TRAVIS_JOB_ID = process.env.TRAVIS_JOB_ID || 'unknown';
```
Comment (XhmikosR) [inline lib/convertLcovToCoveralls.js]: This was also unused AFAICT

File: test/getOptions.js [hunk 3]
```diff
@@ -1,329 +1,358 @@
...
+  let synthetic = false;
 
-  var file = path.join(process.cwd(), '.coveralls.yml'), token, service_name, synthetic = false;
   if (fs.existsSync(file)) {
-    var yaml = require('js-yaml');
-    var coveralls_yml_doc = yaml.safeLoad(fs.readFileSync(yml, 'utf8'));
-    token = coveralls_yml_doc.repo_token;
-    if(coveralls_yml_doc.service_name) {
-      service_name = coveralls_yml_doc.service_name;
+    const coverallsYmlDoc = yaml.safeLoad(fs.readFileSync(file, 'utf8'));
```
Comment (XhmikosR) [inline test/getOptions.js]: This was using `yml`, a non-existent variable. Not sure how it worked before.

File: test/getOptions.js [hunk 4]
```diff
@@ -1,329 +1,358 @@
...
+const testRepoTokenDetection = (sut, done) => {
+  const file = path.join(process.cwd(), '.coveralls.yml');
+  let token;
+  let service_name;
+  let synthetic = false;
 
-  var file = path.join(process.cwd(), '.coveralls.yml'), token, service_name, synthetic = false;
   if (fs.existsSync(file)) {
-    var yaml = require('js-yaml');
-    var coveralls_yml_doc = yaml.safeLoad(fs.readFileSync(yml, 'utf8'));
```
Comment (XhmikosR) [inline test/getOptions.js]: `yml` didn't exist, not sure how it even worked before.

**Comment by XhmikosR** [comment]
I'm open to any comments @nickmerwin. I know style is a personal choice, so I'm willing to make any changes. This PRs apart from using ES6 tries to make the style consistent, but of course I can adapt it to your liking.

File: lib/getOptions.js [hunk 5]
```diff
@@ -148,26 +157,26 @@ var getBaseOptions = function(cb){
...
-        if(coveralls_yaml_conf.service_name) {
+        if (coveralls_yaml_conf.service_name) {
           options.service_name = coveralls_yaml_conf.service_name;
         }
       }
-    } catch(ex){
-      logger.warn("Repo token could not be determined.  Continuing without it." +
-                  "This is necessary for private repos only, so may not be an issue at all.");
+    } catch (_) {
+      logger.warn('Repo token could not be determined.  Continuing without it. ' +
```
Comment (XhmikosR) [inline lib/getOptions.js]: Adds the missing space while at it

File: lib/detectLocalGit.js [hunk 6]
```diff
@@ -1,47 +1,57 @@
-var fs = require('fs');
-var path = require('path');
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
 
 // branch naming only has a few excluded characters, see git-check-ref-format(1)
-var REGEX_BRANCH = /^ref: refs\/heads\/([^?*\[\\~^:]+)$/;
```
Comment (XhmikosR) [inline lib/detectLocalGit.js]: This was an unneeded escape

**Comment by nickmerwin** [comment]
Merged in https://github.com/nickmerwin/node-coveralls/pull/253

**Pull request closed without merging**